controllers/openstackcluster_controller.go

```diff
@@ -206,16 +207,14 @@ func deleteBastion(log logr.Logger, osProviderClient *gophercloud.ProviderClient
 		if err != nil {
 			return err
 		}
+		addresses := instanceNS.Addresses()
 
-		floatingIP := instanceNS.FloatingIP()
-		if floatingIP != "" {
-			if err = networkingService.DisassociateFloatingIP(openStackCluster, floatingIP); err != nil {
-				handleUpdateOSCError(openStackCluster, errors.Errorf("failed to disassociate floating IP: %v", err))
-				return errors.Errorf("failed to disassociate floating IP: %v", err)
-			}
-			if err = networkingService.DeleteFloatingIP(openStackCluster, floatingIP); err != nil {
-				handleUpdateOSCError(openStackCluster, errors.Errorf("failed to delete floating IP: %v", err))
-				return errors.Errorf("failed to delete floating IP: %v", err)
+		for _, address := range addresses {
```

Comment:
just curious, what's the reason behind the removal of disasociate floating ip for bastion only?

Reply:
Consistency. We've never done it for controllers: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/cf5e5cb826db47cf79ccd3d8bdd2a779f3c584c0/controllers/openstackmachine_controller.go#L246-L251

Also it's redundant, because the floating IP will be disassociated automatically on deletion.

Reply:
ok, thanks for the confirmation
I don't know the disassociate also happen on deletion.. thanks:)